Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [22]:
import numpy as np
x = np.array([ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150])
n = 20
N_1 = 20/2
N_2 = (20/2)+1
N_1 = 9 # так как счет начинается с нуля, соответственно на 1 меньше 
N_2 = 10 # так как счет начинается с нуля, соответственно на 1 меньше  


In [23]:
x = np.array([ 17,  24,  25,  30,  33,  45,  55,  57,  65,  65,  70,  75,  75,
        77,  80,  84,  89,  90, 100, 150])
X_sr = (x[9]+x[10])/2

X_sr 

67.5

среднее арифметическое = 67.5
X_sr = 67,5
n = 20
i = 1
Далее находим среднеквадратичное отклонение и смещенные - не смещенные дисперсии.
S^2 = ((x_i - X_sr)^2 + (x_n - X_sr)^2)/n-1

In [19]:

S_difference = sum([(i - X_sr) ** 2 for i in x])#несмещенная дисперсия.
S_difference_1 = S_difference / n
print(f'Несмещенная дисперсия = {S_difference_1}')

S_difference = sum([(i - X_sr) ** 2 for i in x])#смещенная дисперсия.
S_difference_2 = S_difference / n-1
print(f'Смещенная дисперсия = {S_difference_2}')

Несмещенная дисперсия = 954.95
Смещенная дисперсия = 953.95


In [20]:
num_1 = 954.95
num_2 = 953.95
sqrt_1 = num_1 ** (0.5)
sqrt_2 = num_2 ** (0.5)
print("Квадратный корень из числа "+str(num_1)+" это "+str(sqrt_1))
print("Квадратный корень из числа "+str(num_2)+" это "+str(sqrt_2))

Квадратный корень из числа 954.95 это 30.90226528913374
Квадратный корень из числа 953.95 это 30.886081007470015
